### PR TITLE
HPACKHeaders performance test for HTTPHeader normalisation

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

`HPACKHeaders` has an `init` which will normalize `HTTPHeaders`, this
includes removing any headers which are illegal in HTTP2 and any headers
which present as a 'connection' header value. As written this is
expensive in some edges cases. With perf tests we can validate that
implementation changes improve performance.

Modifications:

Add a performance test to normalize `HTTPHeaders`. This is run on two
configurations.

Result:

More insight into header normalisation perf.

### CODE REVIEW

**Comment by glbrntt** [comment]
@swift-nio-bot test perf please

**Comment by glbrntt** [comment]
@swift-nio-bot test perf please

File: Sources/NIOHTTP2PerformanceTester/main.swift [hunk 1]
```diff
@@ -92,6 +92,8 @@ try measureAndPrint(desc: "hpackheaders_canonical_form", benchmark: HPACKHeaderC
 try measureAndPrint(desc: "hpackheaders_canonical_form_trimming_whitespace", benchmark: HPACKHeaderCanonicalFormBenchmark(.trimmingWhitespace))
 try measureAndPrint(desc: "hpackheaders_canonical_form_trimming_whitespace_short_strings", benchmark: HPACKHeaderCanonicalFormBenchmark(.trimmingWhitespaceFromShortStrings))
 try measureAndPrint(desc: "hpackheaders_canonical_form_trimming_whitespace_long_strings", benchmark: HPACKHeaderCanonicalFormBenchmark(.trimmingWhitespaceFromLongStrings))
+try measureAndPrint(desc: "hpackheaders_normalize_httpheaders_removing_10k_conn_headers", benchmark: HPACKHeadersNormalizationOfHTTPHeadersBenchmark(httpHeaders: .manyUniqueConnectionHeaderValues(10_000), iterations: 10))
```
Comment (Lukasa) [inline Sources/NIOHTTP2PerformanceTester/main.swift:95]: Can I ask to move the initialization of these `httpHeaders` values into the `setUp` function? It's a weird thing, but it makes the allocation profilers work better in cases where we only want to see one benchmark.

**Comment by yim-lee** [comment]
@swift-nio-bot test perf please

**Comment by yim-lee** [comment]
@swift-nio-bot test perf please

**Comment by yim-lee** [comment]
Fixed performance jobs and updated them to use Swift 5.6.

**Comment by glbrntt** [comment]
> Fixed performance jobs and updated them to use Swift 5.6.

Thanks Yim!

**Comment by glbrntt** [comment]
@swift-nio-bot test perf please

**Review by Lukasa — APPROVED** [review]

**Pull request merged**